mobile/src/components/setting/setting.tsx

```diff
@@ -1,21 +1,115 @@
 import React, { FC } from 'react';
 
-import { Button, Text, View } from '~/components/common/common';
-import { useAppDispatch } from '~/hooks/hooks';
-import { authActions } from '~/store/actions';
+import defaultUserAvatar from '~/assets/images/avatar-default.png';
+import { ButtonVariant, DataStatus, UserGender } from '~/common/enums/enums';
+import { UserDetailsUpdateInfoRequestDto } from '~/common/types/types';
+import {
+  Button,
+  Dropdown,
+  Image,
+  Input,
+  Spinner,
+  Stack,
+  Text,
+  View,
+} from '~/components/common/common';
+import { getImageUri } from '~/helpers/helpers';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useEffect,
+} from '~/hooks/hooks';
+import { authActions, userDetailsActions } from '~/store/actions';
+import { userDetailsUpdateInfo as userDetailsUpdateInfoValidationSchema } from '~/validation-schemas/validation-schemas';
 
+import {
+  DEFAULT_UPDATE_USER_DETAILS_PAYLOAD,
+  GENDER_OPTIONS,
+} from './common/contants';
 import { styles } from './styles';
 
 const Settings: FC = () => {
   const dispatch = useAppDispatch();
+  const { userDetails, dataStatus } = useAppSelector(
+    (state) => state.userDetails,
+  );
+
+  const { control, errors, handleSubmit, reset } =
+    useAppForm<UserDetailsUpdateInfoRequestDto>({
+      defaultValues: {
+        ...DEFAULT_UPDATE_USER_DETAILS_PAYLOAD,
+      },
+      validationSchema: userDetailsUpdateInfoValidationSchema,
+    });
+
+  const handleGetUsers = (): void => {
+    dispatch(userDetailsActions.getUserDetails());
+  };
```

Comment:
I think you don't need request here, just reset form to the userDetails as in useEffect
```suggestion
  const handleCancel = (): void => {
    // reset form
  };
```